consul/base.py

```diff
@@ -783,6 +786,9 @@ def join(self, address, wan=False):
 
             if wan:
                 params.append(('wan', 1))
+            token = token or self.agent.token
```

Comment:
This action seems to be repeated multiple times

What about creating a helper to do that?

def update_token(self, params, token):
 ... 
 
 And just call self.update_token(params, token) ?

Reply:
Clearly it is, multiple things like that could be factorized in this library.
I took the approach of keeping things as-is for now.
What do you think of doing this in a dedicated PR?

Reply:
ok for me